drivers/ethernet/eth_stm32_hal.c

```diff
@@ -72,6 +167,15 @@ static inline void disable_mcast_filter(ETH_HandleTypeDef *heth)
 {
 	__ASSERT_NO_MSG(heth != NULL);
 
+#if defined(CONFIG_SOC_SERIES_STM32H7X)
+	ETH_MACFilterConfigTypeDef MACFilterConf;
+	HAL_ETH_GetMACFilterConfig(heth, &MACFilterConf);
+	MACFilterConf.HashMulticast = DISABLE;
+	MACFilterConf.PassAllMulticast = ENABLE;
+	MACFilterConf.HachOrPerfectFilter = DISABLE;
+	HAL_ETH_GetMACFilterConfig(heth, &MACFilterConf);
+	k_sleep(K_MSEC(1));
```

Comment:
Why is sleep needed?

Reply:
> Why is sleep needed?

To hold same runtime of the function `disable_mcast_filter` as t ws before. Should it be deleted?

Reply:
Using k_sleep() might cause a context switch to other thread, would that be an issue?
Other alternative is to use `k_busy_wait()` here. The original code is calling `k_sleep()` too so perhaps this is not an issue.

Reply:
@jukkar Since the function `HAL_ETH_SetMACFilterConfig()` was already executed at this time point, the context switch would not be a problem.